test/index.html

```diff
@@ -0,0 +1,247 @@
+        // However, other QUnit.test() blocks than this will fail miserably if it
+        // weren't supported.
+        assert.ok( !document.body.classList.contains("impress-disabled"),
+                   "body.impress-disabled is removed." );
+        assert.ok( document.body.classList.contains("impress-enabled"),
+                   "body.impress-enabled is added." );
+        
+        var canvas = document.querySelector( "div#impress > div" );
+        assert.ok( !canvas.classList.contains("step") && canvas.id === "",
+                   "Additional 'canvas' div inserted between div#impress root and steps." );
+        assert.equal( canvas.style.transform,
+                      "rotateZ(0deg) rotateY(0deg) rotateX(0deg) translate3d(1000px, 0px, 0px)",
+                      "canvas.style.transform initialized correctly" );
+        assert.equal( canvas.style.transformOrigin,
+                      "left top 0px",
+                      "canvas.style.transformOrigin initialized correctly" );
+        assert.equal( canvas.style.transformStyle,
+                      "preserve-3d",
+                      "canvas.style.transformStyle initialized correctly" );
+        assert.equal( canvas.style.transitionDelay,
+                      "0ms",
+                      "canvas.style.transitionDelay initialized correctly" );
+        // impress.js default values tries to set this to 1000ms, I'm completely confused about why that's not actually set in the browser?
+        assert.equal( canvas.style.transitionDuration,
+                      "0ms",
+                      "canvas.style.transitionDuration initialized correctly" );
+        assert.equal( canvas.style.transitionProperty,
+                      "all",
+                      "canvas.style.transitionProperty initialized correctly" );
+        assert.equal( canvas.style.transitionTimingFunction,
+                      "ease-in-out",
+                      "canvas.style.transitionTimingFunction initialized correctly" );
+                      
+        assert.equal( document.documentElement.style.height,
+                      "100%",
+                      "documentElement.style.height is 100%" );
+        
+        // Steps initialization
+        var step1 = document.querySelector( "div#step-1" );
+        assert.equal( step1.style.position,
+                      "absolute",
+                      "Step position is 'absolute'." );
+
+        assert.ok( step1.classList.contains("active"),
+                   "Step 1 has active css class." );
+      }
+    });
+
+    // Note: Here we focus on testing the core functionality of moving between
+    // steps, the css classes set and unset, and events triggered.
+    // TODO: more complex animations and check position, transitions, delays, etc...
+    // Those need to be separate html files, and there could be several of them.
+    QUnit.test( "Impress Core API", function( assert ) {
```

Comment:
Are you referring to the TODO line? I realize now it's unclear (or incomplete, more correctly). My current thinking is that such tests are needed for proper coverage, but they would have to be their own html file. This is because tests operate on the div.step elements we've defined directly under body. So we can only have one impress.js presentation in one html file. The 4 steps in this html file are of course very simple, and allow for basic tests, like testing transitions. To test more advanced animations and positioning, another html file will be needed. But that will still be testing the Core API.